packages/ra-auth-cognito/src/useCognitoLogin.ts

```diff
@@ -54,33 +80,89 @@ export type UseCognitoLoginResult<NewPasswordFormData = unknown> = [
  *     </Form>
  * }
  */
-export const useCognitoLogin = <NewPasswordFormData = unknown>({
+export const useCognitoLogin = ({
     redirectTo,
 }: {
-    redirectTo: string;
-}): UseCognitoLoginResult<NewPasswordFormData> => {
+    redirectTo?: string;
+}): UseCognitoLoginResult => {
     const login = useLogin();
     const [requireNewPassword, setRequireNewPassword] = useSafeSetState(false);
+    const [requireMfaTotp, setRequireMfaTotp] = useSafeSetState(false);
+    const [requireMfaTotpAssociation, setRequireMfaTotpAssociation] =
+        useSafeSetState(false);
+    const [secretCode, setSecretCode] = useSafeSetState('');
+    const [username, setUsername] = useSafeSetState('');
+    const [applicationName, setApplicationName] = useSafeSetState('');
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** I wonder if we could simplify the related code with
```suggestion
    const [applicationResume, setApplicationResume] = useSafeSetState({
        username: '',
        secretCode: '',
        applicationName: '',
    });
```

Reply:
I don't think it's a good idea, it will break custom implementations.